src/components/Table/BatchActionTable.tsx

```diff
@@ -41,7 +43,10 @@ const BatchActionTable = () => {
       };
     });
     setData(updatedData);
-    setAllSelected(updatedData.every((row) => row.selected));
+    const allRowsSelected = updatedData.every((row) => row.selected);
+    const noRowsSelected = updatedData.every((row) => !row.selected);
+    setAllSelected(allRowsSelected);
```

Comment:
From what I can see there is still an issue with the parent checkbox going into allSelected state when manually deselecting all the child checkboxes. It's not connected to Tegel since it's an implementation, but is it something we want to fix?

Reply:
You get different behavior depending on whether you initially pressed parent selection or not, but I think that last commit improved it now.